atrium-core/src/commonMain/kotlin/ch/tutteli/atrium/reporting/text/impl/TextFeatureAssertionGroupFormatter.kt

```diff
@@ -61,6 +61,6 @@ class TextFeatureAssertionGroupFormatter(
         parameterObject: AssertionFormatterParameterObject
     ) : AssertionGroup by assertionGroup {
         override val description: Translatable = newName
-        override val representation: Any = if(parameterObject.isNotInExplanatoryFilterGroup()) assertionGroup.representation else Text.EMPTY
+        override val representation: Any = assertionGroup.representation
```

Comment:
please revert in the end

Reply:
What would be your suggestion then?

Apparently, the current implementation in Atrium is "whatever is displayed in explanatory goes WITHOUT representation".
At the same time, the only way to put "warning sign" is to wrap it in explanatory.

That creates a deadlock, and I think it is wrong that "explanatory" implies "remove all representations". I just do not understand why it removes the representations. I believe the ones who want to display message "without representation" should call the relevant APIs "without representation", and it should not be "explanatory" that hides representations.

Reply:
>  I think it is wrong that "explanatory" implies "remove all representations"

The reason why we remove the representation is because usually we explain what expectation was defined without it being bound to a subject. For instance, the `◆ elements need all: ` part. The expectations shown there are not bound to a specific element in the Iterable.

> At the same time, the only way to put "warning sign" is to wrap it in explanatory.

I see now better where your request for a non-fluent assertionBuilder comes from :)

<hr/>

I see overlapping topics here. I think one way forward would be that we can define that a certain `Assertion` should be shown as is within an explanatory group, i.e. kind of escape the explanatory mechanism. This way we would also only show failing expectations etc. as it was not within an explanatory group. 

Another approach would be that we allow more freely what bullet points are used and then you would not use an explanatory group in your case but just a regular list group with another bullet point. 

Have to go will, think about it later on,

Reply:
We need to take the approach with escaping because otherwise we loose context. for this to work, we need:
- change from NonFiltering count to FilteringStack
- A new GroupAssertionType EscapeNonFiltering which behaves like an invisible group but additionally modifies the stack